File: blurb/app/src/main/java/com/mowdowndevelopments/blurb/ui/story/StoryPagerAdapter.java
package com.mowdowndevelopments.blurb.ui.story;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.mowdowndevelopments.blurb.database.entities.Story;

public class StoryPagerAdapter extends FragmentStateAdapter {

    //TODO Set up data for Adapter
    private Story[] stories;

    public StoryPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (stories != null){
            return StoryFragment.newInstance(stories[position]);
        }
        return null;
    }

    @Override
    public int getItemCount() {
        if (stories != null){
            return stories.length;
        }
        return 0;
    }
}

File: blurb/app/src/main/java/com/mowdowndevelopments/blurb/ui/story/StoryPagerAdapter.kt
package com.mowdowndevelopments.blurb.ui.story

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.mowdowndevelopments.blurb.database.entities.Story

class StoryPagerAdapter(fragmentActivity: FragmentActivity, private val stories: Array<Story>?) : FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        if (stories != null) {
            return StoryFragment.newInstance(stories[position])
        }
        throw IllegalStateException("Attempted to create a fragment from a null set of stories.")
    }

    override fun getItemCount(): Int {
        return stories?.size ?: 0
    }
}